src/vmm_mad/remotes/lib/lxd/mapper/mapper.rb

```diff
@@ -182,15 +184,23 @@ def unmap(one_vm, disk, directory)
         OpenNebula.log_info "Unmapping disk at #{directory}"
 
         sys_parts  = lsblk('')
-        partitions = []
-        device     = ''
 
         return false if !sys_parts
 
+        partitions = []
+
+        df = "#{COMMANDS[:df]} #{directory} | grep /dev"
+        rc, device, err = Command.execute(df, false)
+        device = device.chomp.split(' ')[0]
+
+        if rc != 0
+            OpenNebula.log_error("unmap: #{err}")
+            return
+        end
+
         sys_parts.each { |d|
-            if d['mountpoint'] == directory
+            if d['path'] == device
```

Comment:
Why is this needed, does not lsblk output include the mountpoint? At this point it should (because of at unmap action) When it is not included